testsuite/features/step_definitions/common_steps.rb

```diff
@@ -1187,15 +1187,17 @@ def server_secret
 
 # content lifecycle steps
 When(/^I click the environment build button$/) do
-  raise "Click on environment build failed" unless find(:xpath, '//*[@id="cm-build-modal-save-button"]').click
+  raise 'Click on environment build failed' unless find(:xpath, '//*[@id="cm-build-modal-save-button"]').click
 end
 
 When(/^I click promote from Development to QA$/) do
-  raise "Click on promote from Development failed" unless find(:xpath, '//*[@id="dev_name-promote-modal-link"]').click
+  step %(I wait until button "dev_name-promote-modal-link" becomes enabled)
```

Comment:
I wonder if just to set a "wait: XXX" parameter in the find method, with a big value, could make the same result avoiding one extra step.

Reply:
It sounds good; I am going to investigate it.